### PR TITLE
[WFLY-11003] Add documentation describing how to define an SSLContextfor SNI in the WildFly Elytron subsystem.

### PR DESCRIPTION
**Pull request by darranl**
Thanks for submitting your Pull Request!

Please make sure your PR meets the following requirements:
- [x] Pull Request title is properly formatted: `[WFLY-XYZ] Subject` or `WFLY-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue(s)

https://issues.jboss.org/browse/WFLY-11003

- [x] Pull Request contains description of the issue(s)

This pull request is adding documentation for the SNI support added under PR: -
  https://github.com/wildfly/wildfly-core/pull/3513

- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Attached commits represent units of work and are properly formatted

For bigger changes, major and minor component upgrades make sure your PR also meets following requirements:
- [x] Pull Request requires a change to the documentation
- [x] Documentation have been updated accordingly
- [ ] Tests were added to cover changes

For new features ensure as well:
- [x] Analysis was done
- [ ] Test Plan has been done
- [ ] Tests were verified in advance

If you are not an active contributor of the WildFly project you can request sponsorship by one of the members to help guide you through the process.

### CODE REVIEW

File: docs/src/main/asciidoc/_elytron/Using_the_Elytron_Subsystem.adoc [hunk 1]
```diff
@@ -1999,6 +1999,41 @@ use custom implementations of the following components:
 When creating custom implementations of Elytron components, they must
 present the appropriate capabilities and requirements.
 
+=== Configuring SNI
+
+Using the WildFly Elytron subsystem it is possible to configure an SSL context which supports SNI.  By supporting SNI if an SNI host name is available whilst the SSLSession is being negotiation a host specific SSLcontext will be selected.  If no host specific SSLContext is identified either because no host name was received or because there is no match a default SSLContext will be used instead.  By identifying a host specific SSLContext it means that a certificate appropriate for that host can be used.
```
Comment (jamezp) [inline docs/src/main/asciidoc/_elytron/Using_the_Elytron_Subsystem.adoc]: Minor detail but
> host specific SSLcontext

Should probably be `SSLContext` to match the rest of the case.

Comment (bstansberry) [reply docs/src/main/asciidoc/_elytron/Using_the_Elytron_Subsystem.adoc]: also in that same clause...

s/is being negotation/is being negotiated/g

**Comment by darranl** [comment]
@jamezp Can you please remove the 'rebase-this' label - this one is ready to go in.

**Review by jamezp — APPROVED** [review]

**Comment by jamezp** [comment]
I killed the CI jobs since it's only a doc change.

File: docs/src/main/asciidoc/_elytron/Using_the_Elytron_Subsystem.adoc [hunk 2]
```diff
@@ -2038,6 +2038,41 @@ As existing services and deployments could have cached the default `SSLContext`
...
+
+[source, ruby]
+----
+[standalone@localhost:9990 /] ./subsystem=elytron/server-ssl-sni-context=test-sni:add(default-ssl-context=jboss,host-context-map={localhost=localhost, wildfly.org=wildfly})
+{"outcome" => "success"}
+----
+
+This example assumes that three SSLContexts have been previously defined following the steps available previously in this document, those contexts are `jboss`, `localhost`, and `wildfly`.
+
+During negotiation of the SSLSession if the SNI host name received is `localhost` then the `localhost` SSLContext will be used, if the SNI host name is `wildfly.org` then the `wildfly` SSLContext will be used.  If no SNI host name is received or if we receive a name that does not match this will fallback and use the `jboss` SSLContext.
```
Comment (jstourac) [inline docs/src/main/asciidoc/_elytron/Using_the_Elytron_Subsystem.adoc]: Actually, as `.` (dot) character in the mapping is used as a reg-exp character, then `wildfly.org` will match e.g.: `wildflyLorg`, `wildflyborg`, etc., see https://issues.jboss.org/browse/WFWIP-102.

**Pull request merged**